src/web/src/alert_ack.ecpp

```diff
@@ -67,205 +79,174 @@ bool database_ready;
-        log_debug ("State is not a recognized valid alert state.");
-        std::string err = TRANSLATE_ME ("one of the following values [ ACTIVE | ACK-WIP | ACK-IGNORE | ACK-PAUSE | ACK-SILENCE ].");
-        http_die ("request-param-bad", "state", state.c_str (), err.c_str ());
-    }
-    else {
-        checked_state = state;
-    }
         http_errors_t errors;
         if (! check_asset_name ("element_name", element_name, errors)) {
             http_die_error (errors);
         }
         checked_element_name = element_name;
+
         if (! check_alert_rule_name ("rule_name", rule_name, errors)) {
             http_die_error (errors);
         }
         checked_rule_name = rule_name;
-}
-
 
-// connect to malamute
-MlmClientGuard client(mlm_client_new ());
-if (!client) {
-    log_fatal ("mlm_client_new () failed.");
-    std::string err = TRANSLATE_ME ("mlm_client_new () failed.");
-    http_die ("internal-error", err.c_str ());
-}
-
-std::string client_name = utils::generate_mlm_client_id ("web.alert_ack");
-log_debug ("malamute client name = '%s'.", client_name.c_str ());
-
-int rv = mlm_client_connect (client, MLM_ENDPOINT, 1000, client_name.c_str ());
-if (rv == -1) {
-    log_fatal ("mlm_client_connect (endpoint = '%s', timeout = '%d', address = '%s') failed.",
-                    MLM_ENDPOINT, 1000, client_name.c_str ());
-    std::string err =  TRANSLATE_ME ("mlm_client_connect () failed.");
-    http_die ("internal-error", err.c_str ());
-}
-
-// prepare rfc-evaluator-rules ADD message
-zmsg_t *send_msg = zmsg_new ();
-if (!send_msg) {
-    log_fatal ("zmsg_new () failed.");
-    std::string err =  TRANSLATE_ME ("zmsg_new () failed.");
-    http_die ("internal-error", err.c_str ());
-}
-std::string asset_id;
-rv = DBAssets::extname_to_asset_name (checked_element_name, asset_id);
-if (rv == -1) {
-    std::string msg_not_found = TRANSLATE_ME ("Cannot get asset ID for %s", checked_element_name.c_str ());
-    http_die ("not-found", msg_not_found.c_str ());
-}
-if (rv == -2) {
-    std::string err =  TRANSLATE_ME ("Database failure");
-    http_die ("internal-error", err.c_str ());
-}
-zmsg_addstr (send_msg, checked_rule_name.c_str ());
-zmsg_addstr (send_msg, asset_id.c_str ());
-zmsg_addstr (send_msg, checked_state.c_str ());
+        // validate request document
+        std::string state;
+        cxxtools::SerializationInfo si;
+        try {
+            std::stringstream input (request.getBody (), std::ios_base::in);
+            cxxtools::JsonDeserializer deserializer (input);
+            deserializer.deserialize (si);
+            si.getMember ("state") >>= state;
+            if (state.empty ())
+                throw std::runtime_error (TRANSLATE_ME ("property 'state' has empty value"));
+        }
+        catch (const std::exception& e) {
+            log_debug ("Bad request document - invalid json: %s", e.what ());
+            std::string err = TRANSLATE_ME ("Expected format: { \"state\" : <state> }");
```

Comment:
@jeffdegott , You are commenting part of code which was deleted, new message is here :-)

Reply:
uuuh ;)